include/openssl/x509.h

```diff
@@ -510,18 +510,18 @@ EVP_PKEY *X509_PUBKEY_get0(X509_PUBKEY *key);
 EVP_PKEY *X509_PUBKEY_get(X509_PUBKEY *key);
 int X509_get_pubkey_parameters(EVP_PKEY *pkey, STACK_OF(X509) *chain);
 long X509_get_pathlen(X509 *x);
-int i2d_PUBKEY(EVP_PKEY *a, unsigned char **pp);
+int i2d_PUBKEY(const EVP_PKEY *a, unsigned char **pp);
 EVP_PKEY *d2i_PUBKEY(EVP_PKEY **a, const unsigned char **pp, long length);
```

Comment:
Hmmm...

``` C
DECLARE_ASN1_ENCODE_FUNCTIONS_const(EVP_PKEY, PUBKEY);
```

Reply:
Well, it expands to
```
 OPENSSL_EXTERN const ASN1_ITEM name##_it;
```
which seems fine in most cases, but I've just found cases where (for whatever reason) the `_it` function is static, and then `DECLARE_ASN1_ENCODE_FUNCTIONS_const` (and `DECLARE_ASN1_ENCODE_FUNCTIONS`) cannot be used in header files. I'll continue on this tomorrow.

Reply:
To be on the safe side and to avoid the mentioned issue with static `_it` functions, I've introduced and used instead `DECLARE_ASN1_ENCODE_FUNCTIONS_only`. 
BTW, `DECLARE_ASN1_ENCODE_FUNCTIONS_const` is not needed any more, see my extra comment below.